src/bridge.ts

```diff
@@ -1524,10 +1525,8 @@ export class Bridge {
 
         metrics.registerMatrixSdkMetrics(this.botSdkAS);
 
-        // TODO(paul): register some bridge-wide standard ones here
-
-        // In case we're called after .run()
-        if (this.appService && registerEndpoint) {
+        // This will throw if true and not called after run.
```

Comment:
I wonder if we want to throw early if we know that it's gonna happen. Two layers from here we'll get a `Error('Cannot call addAppServicePath before calling .run()')`, and in this context the user is calling `getPrometheusMetrics` rather than `addAppServicePath` before calling `listen()` rather than `run()`.

Ideally I'd see all of these methods that require listen to be called first to be moved to a separate object that gets returned from listen(), so that API itself would prevent you from doing it wrong, but maybe that's an undertaking big enough for a 4.0.0 ;)

Reply:
Yeah, definitely a bad smell that we have these at all but I think best reserved for 4.0.0.